In [1]:
import folium
import requests
import pandas as pd
import re

In [2]:
crash_table = pd.read_csv("Maryland_Statewide_Vehicle_Crashes.csv")
crash_table.head()

/tmp/ipykernel_167/2940651411.py:1: DtypeWarning: Columns (34,46) have mixed types. Specify dtype option on import or set low_memory=False.
  crash_table = pd.read_csv("Maryland_Statewide_Vehicle_Crashes.csv")


YEAR QUARTER LIGHT_DESC  LIGHT_CODE     COUNTY_DESC  COUNTY_NO  MUNI_DESC  \
0  2020      Q2   Daylight        1.00       Baltimore        3.0        NaN   
1  2020      Q2        NaN        6.02  Baltimore City       24.0        NaN   
2  2020      Q2   Daylight        1.00      Montgomery       15.0        NaN   
3  2017      Q2   Daylight        1.00  Baltimore City       24.0        NaN   
4  2020      Q2   Daylight        1.00           Cecil        7.0        NaN   

   MUNI_CODE     JUNCTION_DESC  JUNCTION_CODE  ... FEET_MILES_FLAG_DESC  \
0        NaN               NaN            NaN  ...                  NaN   
1        NaN  Non Intersection            1.0  ...                Miles   
2        NaN               NaN            NaN  ...                  NaN   
3        NaN               NaN            NaN  ...                  NaN   
4        NaN               NaN            NaN  ...                  NaN   

   FEET_MILES_FLAG DISTANCE_DIR_FLAG  REFERENCE_NO REFERENCE_TYPE_CODE  \
0              NaN               NaN           NaN                 NaN   
1                M                 N           NaN                 NaN   
2              NaN               NaN           NaN                 NaN   
3              NaN               NaN           NaN                 NaN   
4              NaN               NaN           NaN                 NaN   

   REFERENCE_SUFFIX REFERENCE_ROAD_NAME   LATITUDE  LONGITUDE  \
0               NaN                 NaN  39.277263 -76.503693   
1               NaN           NORTH AVE  39.311025 -76.616429   
2               NaN                 NaN  39.140680 -77.193413   
3               NaN                 NaN  39.282928 -76.635215   
4               NaN                 NaN  39.611028 -75.951314   

                                     LOCATION  
0             POINT (-76.5036932 39.27726285)  
1    POINT (-76.616429453205 39.311024794431)  
2    POINT (-77.193412729561 39.140680249069)  
3  POINT (-76.6352150952347 39.2829284750108)  
4          POINT (-75.951314 39.611027833333)  

[5 rows x 56 columns]

In [3]:
crash_table[pd.notnull(crash_table["LOCATION"])]["LOCATION"].count()

771145

In [4]:
map_osm = folium.Map(location=[39.29, -76.61], zoom_start=13)
map_osm

In [9]:
crash_table.columns

Index(['YEAR', 'QUARTER', 'LIGHT_DESC', 'LIGHT_CODE', 'COUNTY_DESC',
       'COUNTY_NO', 'MUNI_DESC', 'MUNI_CODE', 'JUNCTION_DESC', 'JUNCTION_CODE',
       'COLLISION_TYPE_DESC', 'COLLISION_TYPE_CODE', 'SURF_COND_DESC',
       'SURF_COND_CODE', 'LANE_DESC', 'LANE_CODE', 'RD_COND_DESC',
       'RD_COND_CODE', 'RD_DIV_DESC', 'RD_DIV_CODE', 'FIX_OBJ_DESC',
       'FIX_OBJ_CODE', 'REPORT_NO', 'REPORT_TYPE', 'WEATHER_DESC',
       'WEATHER_CODE', 'ACC_DATE', 'ACC_TIME', 'LOC_CODE', 'SIGNAL_FLAG_DESC',
       'SIGNAL_FLAG', 'C_M_ZONE_FLAG', 'AGENCY_CODE', 'AREA_CODE',
       'HARM_EVENT_DESC1', 'HARM_EVENT_CODE1', 'HARM_EVENT_DESC2',
       'HARM_EVENT_CODE2', 'RTE_NO', 'ROUTE_TYPE_CODE', 'RTE_SUFFIX',
       'LOG_MILE', 'LOGMILE_DIR_FLAG_DESC', 'LOGMILE_DIR_FLAG',
       'MAINROAD_NAME', 'DISTANCE', 'FEET_MILES_FLAG_DESC', 'FEET_MILES_FLAG',
       'DISTANCE_DIR_FLAG', 'REFERENCE_NO', 'REFERENCE_TYPE_CODE',
       'REFERENCE_SUFFIX', 'REFERENCE_ROAD_NAME', 'LATITUDE', 'LONGITUDE',
       'L

In [7]:
for index, crash in crash_table.iterrows():
    if crash["ACC_DATE"] == re.compile("^2021[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'red')).add_to(map_osm)
    elif crash["ACC_DATE"] == re.compile("^2020[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'blue')).add_to(map_osm)
    elif crash["ACC_DATE"] == re.compile("^2019[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'yellow')).add_to(map_osm)
    elif crash["ACC_DATE"] == re.compile("^2018[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'green')).add_to(map_osm)
    elif crash["ACC_DATE"] == re.compile("^2017[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'purple')).add_to(map_osm)
    elif crash["ACC_DATE"] == re.compile("^2016[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'orange')).add_to(map_osm)
    elif crash["ACC_DATE"] == re.compile("^2015[0-9]{4}$"):
        folium.Marker(location = [crash["LATITUDE"], crash["LONGITUDE"]], icon = folium.Icon(color = 'pink')).add_to(map_osm)

map_osm

I choose this dataset "Maryland_Statewide_Vehicle_Crashes" to have the bigger vision for the whole state's vehicle crashes information compared to the example given in the project description. Besides, this dataset has thorough information including 56 columns, including the most important ones "LATITUDE" and "LONGITUDE". This will allow me to analyze those vehicle crashes in a more centralized format, such as here I analyze the all 771145 vehicle crashes over years.

My analyzing logic for this is to differentiate the difference between what year that all the vehicle crashes happened in Maryland statewide (in our dataset).

As shown in my interactive map above, with all 771145 marks drawn in the map, we can tell two main discoveries based on this.

First, by the majority (or the domination) of one color in certain area, we can tell that the frequency of car crashes happened at most in which year, which would possibly give us some inspiration of other things because of this.

Second, by the numbers of colors in overall, we can get that the car crashes happened at most at year 2020, which shown in the map by having the blue color marks at most.